rewrite-core/src/main/java/org/openrewrite/OptionGroup.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite;
+
+import com.fasterxml.jackson.annotation.JacksonAnnotationsInside;
+import com.fasterxml.jackson.annotation.JsonUnwrapped;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Nested options are used to configure recipes.
+ * <p>
+ * They allow for options to be re-used across multiple {@link Recipe} implementations through object composition.
+ * <p>
+ * Fields annotated with this method <i>may</i> also be annotated with {@link JsonUnwrapped} for further configuration
+ * of the deserialized JSON.
+ */
+@Incubating(since = "8.12.0")
+@Target({ElementType.FIELD, ElementType.METHOD})
+@Retention(RetentionPolicy.RUNTIME)
+@JsonUnwrapped
+@JacksonAnnotationsInside
+public @interface OptionGroup {
+}
```

Comment:
I like this idea too, @JLLeitschuh. Perhaps you could also add a few fields to it:
```suggestion
import com.fasterxml.jackson.annotation.JacksonAnnotationsInside;
import org.intellij.lang.annotations.Language;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Nested options are used to configure recipes.
 * <p>
 * They allow for options to be re-used across multiple {@link Recipe} implementations through object composition.
 * <p>
 * Fields annotated with this method <i>may</i> also be annotated with
 * {@link com.fasterxml.jackson.annotation.JsonProperty} or {@link com.fasterxml.jackson.annotation.JsonUnwrapped}
 * for further configuration of the deserialized JSON.
 */
@Incubating(since = "8.12.0")
@Target({ElementType.FIELD, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
@JacksonAnnotationsInside
public @interface OptionGroup {
    @Language("markdown")
    String displayName() default "";
    @Language("markdown")
    String description() default "";
    boolean required() default true;
}
```

Reply:
Could you elaborate on what the behavior of those additional arguments would be?

The sub elements of the @OptionGroup will already have those fields on them. Would these be prefixes or suffexes for the children items?

Reply:
This is how I would use it, but I would also add it at least to maintain consistency with existing `@Option`s.

One detail that I think is missing in your implementation is the validation of nested fields.

I removed my comments to keep the review clean, but please let me know if I can help.

Reply:
The goal was to, out of the box, maintain consistent support with the Moderne SaaS. If `@OptionGroup` required specifying nested json/yaml fields, then the Moderne team would need to make changes to the SaaS to support it, which I wanted to avoid

Reply:
This is why I chose to have `@OptionGroup` contain `@JsonUnwrap` by default

Reply:
> One detail that I think is missing in your implementation is the validation of nested fields.

Can you elaborate? AFAIK, I'm not doing anything different from existing recipe validation. If I'm wrong, can you point me towards where I should be looking?

Reply:
@sambsnyd @knutwannheden what are your thoughts here. My intention was to add this annotation such that it required no SaaS changes to support.

Would you prefer to be able to actually support nested JSON inputs OR flattened inputs. Or would you prefer to go with what I have currently which is that `@OptionGroup` always flattens the inputs